doc/source/set_operations.rst

```diff
@@ -0,0 +1,75 @@
+.. ipython:: python
+   :suppress:
+
+   import geopandas as gpd
+   world = gpd.GeoDataFrame().from_file('_example_data/naturalearth_lowres.shp')
+   capitals = gpd.GeoDataFrame().from_file('_example_data/naturalearth_cities.shp')
+   
+   # For spatial join
+   countries = world[['geometry', 'name']]
+
+   # Project
+   countries = countries.to_crs('+init=epsg:3395')[countries.name!="Antarctica"]
+   capitals = capitals.to_crs('+init=epsg:3395')
+
+
+Set-Operations with Overlay
+============================
+
+When working with multiple spatial datasets -- especially multiple *polygon* or *line* datasets -- users often wish to create new shapes based on places where those datasets overlap (or don't overlap). These manipulations are often referred using the language of sets -- intersections, unions, and differences. These types of operations are made available in the *geopandas* library through the ``overlay`` function. 
+
+In particular, ``overlay`` can be used to create new shapes using four different relationships:
+
+.. image:: _static/overlay_operations.png
+
+(Note to users familiar with the *shapely* library: ``overlay`` can be thought of as offering versions of the standard *shapely* set-operations that deal with the complexities of applying set operations to two *GeoSeries*. The standard *shapely* set-operations are also available as ``GeoSeries`` methods.)
```

Comment:
It would maybe be nice to first also give some examples of this (but does not need to be done in this PR)